lib/secretsscanner/scaner/scan.go

```diff
@@ -132,15 +154,20 @@ var (
 )
 
 // readFileIfSSHPrivateKey checks if a file is an OpenSSH private key
-func readFileIfSSHPrivateKey(filePath string) ([]byte, bool, error) {
+func (s *Scanner) readFileIfSSHPrivateKey(ctx context.Context, filePath string) ([]byte, bool, error) {
 	file, err := os.Open(filePath)
 	if err != nil {
 		return nil, false, err
 	}
-	defer file.Close()
+	defer func() {
+		if err = file.Close(); err != nil {
+			s.log.WarnContext(ctx, "failed to close file", "path", filePath, "error", err)
```

Comment:
Suggestion: Log at Debug? A Close error after a read doesn't seem that important.

```suggestion
			s.log.DebugContext(ctx, "failed to close file", "path", filePath, "error", err)
```